src/codegen/llvm.zig

```diff
@@ -10413,11 +10408,20 @@ fn firstParamSRet(fn_info: Type.Payload.Function.Data, target: std.Target) bool
             .riscv32, .riscv64 => return riscv_c_abi.classifyType(fn_info.return_type, target) == .memory,
             else => return false, // TODO investigate C ABI for other architectures
         },
+        .SysV => return firstParamSRetSystemV(fn_info.return_type, target),
+        .Win64 => return x86_64_abi.classifyWindows(fn_info.return_type, target) == .memory,
         .Stdcall => return !isScalar(fn_info.return_type),
         else => return false,
     }
 }
 
+fn firstParamSRetSystemV(ty: Type, target: std.Target) bool {
```

Comment:
Do you think it'd be worth moving this function into `abi.zig` as we will most likely need to do the same for self-hosted backends?

Reply:
It might be but that can be done when implementing `f80` for the x86 backend.